src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenTuples.vb

```diff
@@ -137,6 +137,163 @@ hello
         End Sub
 
         <Fact>
+        Public Sub TupleTypeBindingTypeChar()
+
+            Dim verifier = CompileAndVerify(
+<compilation>
+    <file name="a.vb">
+option strict on
+
+Imports System
+Module C
+
+    Sub Main()
+        Dim t as (A%, B$) = Nothing
+        console.writeline(t.GetType())
+    End Sub
+End Module
+
+    </file>
+</compilation>, additionalRefs:=s_valueTupleRefs, expectedOutput:=<![CDATA[
+System.ValueTuple`2[System.Int32,System.String]
+            ]]>)
+
+            verifier.VerifyDiagnostics()
+
+            verifier.VerifyIL("C.Main", <![CDATA[
+{
+  // Code size       25 (0x19)
+  .maxstack  1
+  .locals init (System.ValueTuple(Of Integer, String) V_0) //t
+  IL_0000:  ldloca.s   V_0
+  IL_0002:  initobj    "System.ValueTuple(Of Integer, String)"
+  IL_0008:  ldloc.0
+  IL_0009:  box        "System.ValueTuple(Of Integer, String)"
+  IL_000e:  call       "Function Object.GetType() As System.Type"
+  IL_0013:  call       "Sub System.Console.WriteLine(Object)"
+  IL_0018:  ret
+}
+]]>)
+        End Sub
+
+        <Fact>
+        Public Sub TupleTypeBindingTypeChar1()
+
+            Dim verifier = CompileAndVerify(
+<compilation>
+    <file name="a.vb">
+option strict off
+
+Imports System
+Module C
+
+    Sub Main()
+        Dim t as (A%, B$) = Nothing
+        console.writeline(t.GetType())
+    End Sub
+End Module
+
+    </file>
+</compilation>, additionalRefs:=s_valueTupleRefs, expectedOutput:=<![CDATA[
+System.ValueTuple`2[System.Int32,System.String]
+            ]]>)
+
+            verifier.VerifyDiagnostics()
+
+            verifier.VerifyIL("C.Main", <![CDATA[
+{
+  // Code size       25 (0x19)
+  .maxstack  1
+  .locals init (System.ValueTuple(Of Integer, String) V_0) //t
+  IL_0000:  ldloca.s   V_0
+  IL_0002:  initobj    "System.ValueTuple(Of Integer, String)"
+  IL_0008:  ldloc.0
+  IL_0009:  box        "System.ValueTuple(Of Integer, String)"
+  IL_000e:  call       "Function Object.GetType() As System.Type"
+  IL_0013:  call       "Sub System.Console.WriteLine(Object)"
+  IL_0018:  ret
+}
+]]>)
+        End Sub
+
+        <Fact>
+        Public Sub TupleTypeBindingTypeCharErr()
+
+            Dim comp = CreateCompilationWithMscorlibAndVBRuntime(
+<compilation name="TupleTypeBindingTypeCharErr">
+    <file name="a.vb"><![CDATA[
+Imports System
+Module C
+
+    Sub Main()
+        Dim t as (A% As String, B$ As String, C As String$) = nothing
+        console.writeline(t.A.Length)  'A should not take the type from % in this case
+
+        Dim t1 as (String$, String%) = nothing
+        console.writeline(t1.GetType())
+
+        Dim B = 1
+        
+        Dim t2 = (A% := "qq", B$) 
+        console.writeline(t2.A.Length) 'A should not take the type from % in this case
+
+        Dim t3 As (V1(), V2%()) = Nothing
+        console.writeline(t3.Item1.Length) 
+    End Sub
+
+   Async Sub T()
+        Dim t4 as (Integer% As String, Await As String, Function$) = nothing
```

Comment:
Yes and it is more interesting since it is an identifier. Keywords are easier since they could not be element names.
Added couple scenarios.

---
In reply to: [82285387](https://github.com/dotnet/roslyn/pull/14284#discussion_r82285387) [](ancestors = 82285387)